### ISSUE TITLE
[feat] new `--exclude` flag for `flutter analyze`

### ISSUE THREAD

**Issue by opxdelwin** [flutter/flutter#141928]
Edit by @christopherfujino, this is currently blocked by https://github.com/dart-lang/sdk/issues/54826

---

### Use case

Rather than excluding a certain file from analysis for the entire project, We can enable the devs to exclude certain file from analysis only for a single instance of `flutter analyze`.

Ran into this trouble where I wanted to exclude one file from analyze when done via github actions, as certain firebase_config files aren't in the repo, but kept locally.

### Proposal

Create a new flag `--exclude <relative file location>` for `flutter analyze` to exclude this file for this specific `flutter analyze` run

**Comment by christopherfujino** [comment]
@opxdelwin this sounds reasonable to me. However, I don't think we would actually implement this in the Flutter CLI tool--since the actual analysis happens upstream in the dart analysis server (https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/dart/analysis.dart#L61), and (to my knowledge) the dart analysis server command line options do not support this feature yet: https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/lib/src/server/driver.dart#L717

In other words, I think the way forward would be to file a feature request on https://github.com/dart-lang/sdk/issues/new/choose for this CLI flag to be implemented on the analysis server. If that were implemented, we could then trivially forward it from Flutter.

**Comment by christopherfujino** [comment]
> Filed [dart-lang/sdk#54826](https://github.com/dart-lang/sdk/issues/54826)

Thanks!
